ProjectLighthouse.Servers.Website/Controllers/Admin/AdminUserController.cs

```diff
@@ -91,6 +93,63 @@ public AdminUserController(DatabaseContext database)
 
         return this.Redirect($"/user/{targetedUser.UserId}");
     }
+    
+    /// <summary>
+    /// Deletes every comment by the user. Useful in case of mass spam
+    /// </summary>
+    [HttpGet("wipeComments")]
+    public async Task<IActionResult> WipeComments([FromRoute] int id) {
+        UserEntity? user = this.database.UserFromWebRequest(this.Request);
+        if (user == null || !user.IsModerator) return this.NotFound();
+
+        UserEntity? targetedUser = await this.database.Users.FirstOrDefaultAsync(u => u.UserId == id);
+        if (targetedUser == null) return this.NotFound();
+
+        // Get every comment posted by the target user
+        List<CommentEntity> comments = await this.database.Comments
+            .Where(c => c.Poster == targetedUser)
```

Comment:
This might be some premature optimization but EntityFramework has support for bulk updates which can be done like this
```csharp
await database.Comments.Where(c => c.PosterUserId == targetUser.UserId)
            .ExecuteUpdateAsync(s =>
                   s.SetProperty(c => c.Deleted, true)
                    .SetProperty(c => c.DeletedBy, user.Username)
                    .SetProperty(c => c.DeletedType, "moderator"));
```
I think the for loop is more readable IMO but you would have to performance test to see how each method would scale.

Reply:
i dunno, but i like this better anyway